apps/web/src/ui/components/commons/search-component/index.tsx

```diff
@@ -0,0 +1,32 @@
+import { Input } from '@nextui-org/react'
+import { Search } from 'lucide-react'
+import { useState, useCallback, useMemo } from 'react'
+
+interface SearchComponentsProps {
+  onSearchChange: (value: string) => void
+  filterByNameValue: string
+}
+
+export const TableSearchComponent = ({
+  onSearchChange,
+  filterByNameValue,
+}: SearchComponentsProps) => {
+  return useMemo(() => {
```

Comment:
`useMemo` is completely desnecessary here. this hook usually used to memoize heavy calculations only

Reply:
there will be when an array of 1000+ products is being filtered letter by letter